dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -604,5 +622,128 @@ public String getUrl(Contentlet contentlet) {
+        }
+    }
+
+    @DataProvider
+    public static Object[] listSerializeTestCases() throws Exception {
+
+        final FileAssetAPI assetAPI = APILocator.getFileAssetAPI();
+        final HTMLPageAssetAPI pageAssetAPI = APILocator.getHTMLPageAssetAPI();
+
+        return new Object[]{
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getFileAssetSVGContent(true, langId);
+                    return assetAPI.fromContentlet(contentlet);
+                }).get(), fileAssetValidation),
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getFileAssetContent(true, langId);
+                    return assetAPI.fromContentlet(contentlet);
+                }).get(), fileAssetValidation),
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getPageContent(true, langId);
+                    return pageAssetAPI.fromContentlet(contentlet);
+                }).get(), pageValidation),
+                new SerializationTestCase(TestDataUtils.getDotAssetLikeContentlet(), null),
+                new SerializationTestCase(TestDataUtils.getBlogContent(true, langId), null)
+        };
+    }
+
+    private void validateTransformation(final Contentlet original, final Contentlet copy){
+        final DotContentletTransformer transformer1 = new DotTransformerBuilder().defaultOptions().content(original).build();
+        final Map<String, Object> preSerializationMap = transformer1.toMaps().get(0);
+
+        final DotContentletTransformer transformer2 = new DotTransformerBuilder().defaultOptions().content(copy).build();
+        final Map<String, Object> postSerializationMap = transformer2.toMaps().get(0);
+        Assert.assertEquals(preSerializationMap, postSerializationMap);
+
+        Assert.assertEquals(transformer1.hydrate().get(0), transformer2.hydrate().get(0));
+    }
+
+    private static final AssertionStrategy fileAssetValidation = (original, copy) -> {
+        final FileAsset fileAsset1 = (FileAsset) original;
+        final int width = fileAsset1.getWidth();
+        final int height = fileAsset1.getHeight();
+        final FileAsset fileAsset2 = (FileAsset) copy;
+        final int width2 = fileAsset2.getWidth();
+        final int height2 = fileAsset2.getHeight();
+        Assert.assertEquals(width, width2);
+        Assert.assertEquals(height, height2);
+    };
+
+    private static final AssertionStrategy pageValidation = (original, copy) -> {
+        final HTMLPageAsset htmlPageAsset1 = (HTMLPageAsset) original;
+        final HTMLPageAsset htmlPageAsset2 = (HTMLPageAsset) copy;
+        Assert.assertEquals(htmlPageAsset1.getPageUrl(),htmlPageAsset2.getPageUrl());
+        Assert.assertEquals(htmlPageAsset1.getRedirect(),htmlPageAsset2.getRedirect());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5883765)